src/essreflectometry/amor/orso.py

```diff
@@ -1,60 +1,79 @@
-
-    Returns
-    -------
-    :
-        Orso object with the default parameters for the Amor instrument.
-    """
-    orso = fileio.orso.Orso.empty()
-    orso.data_source.experiment.probe = 'neutrons'
-    orso.data_source.experiment.facility = 'Paul Scherrer Institut'
-    orso.data_source.measurement.scheme = 'angle- and energy-dispersive'
-    orso.reduction.software = fileio.reduction.Software(
-        'scipp-ess', __version__, platform.platform()
+"""ORSO utilities for Amor."""
+import os
+from typing import Optional
+
+from dateutil.parser import parse as parse_datetime
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source
+
+from ..orso import (
+    OrsoExperiment,
+    OrsoInstrument,
+    OrsoMeasurement,
+    OrsoOwner,
+    OrsoSample,
+)
+from ..types import Filename, RawData, Reference, Run, Sample
+
+
+def parse_orso_experiment(raw_data: RawData[Run]) -> OrsoExperiment[Run]:
+    """Parse ORSO experiment data from raw Amor NeXus data."""
+    return OrsoExperiment(
+        data_source.Experiment(
+            title=raw_data['title'],
+            instrument=raw_data['name'],
+            facility=raw_data['facility'],
+            start_date=parse_datetime(raw_data['start_time']),
+            probe='neutron',
+        )
+    )
+
+
+def parse_orso_owner(raw_data: RawData[Run]) -> OrsoOwner[Run]:
+    """Parse ORSO owner data from raw Amor NeXus data."""
+    return OrsoOwner(
+        orso_base.Person(
+            name=raw_data['user']['name'],
+            contact=raw_data['user']['email'],
+            affiliation=None,
+        )
     )
-    orso.reduction.timestep = datetime.now()
-    orso.reduction.corrections = []
-    orso.reduction.computer = platform.node()
-    orso.columns = [
-        fileio.base.Column('Qz', '1/angstrom', 'wavevector transfer'),
-        fileio.base.Column('R', None, 'reflectivity'),
-        fileio.base.Column('sR', None, 'standard deivation of reflectivity'),
-        fileio.base.Column(
-            'sQz', '1/angstrom', 'standard deviation of wavevector transfer resolution'
-        ),
-    ]
-    orso.data_source.owner = owner
-    orso.data_source.sample = sample
-    orso.reduction.creator = creator
-    orso.reduction.script = reduction_script
-    return orso
+
+
+def parse_orso_sample(raw_data: RawData[Run]) -> OrsoSample[Run]:
+    """Parse ORSO sample data from raw Amor NeXus data."""
+    if not raw_data.get('sample'):
+        return OrsoSample(data_source.Sample.empty())
+    raise NotImplementedError('Amor NsXus sample parsing is not implemented')
+
+
+def build_orso_measurement(
+    sample_filename: Filename[Sample],
+    reference_filename: Filename[Reference],
+    instrument: Optional[OrsoInstrument],
+) -> OrsoMeasurement:
+    """Assemble ORSO measurement data."""
+    # TODO populate timestamp
+    #      doesn't work with a local file because we need the timestamp of the original,
+    #      SciCat can provide that
```

Comment:
What is the difference to the experiment time above? Isn't the time in NXentry actually a measurement time, not an experiment time?

Reply:
> What is the difference to the experiment time above?

There is only one experiment time (sample run) but we have multiple files that each have a creation time.

> Isn't the time in NXentry actually a measurement time, not an experiment time?

What is the difference?

Reply:
I do not know how they define it, but an experiment could consist of multiple measurements (e.g., many samples).

Reply:
Then we need to extend how we handle this here. E.g., use the `max` of all times in sample files. 

We can change this here to use the time recorded in the NXentry instead of the creation time of the file. These are not the same but, as long as there is only one entry, they are essentially equivalent.